main/src/ca/uwaterloo/flix/language/phase/Monomorpher.scala

```diff
@@ -763,25 +750,18 @@ object Monomorpher {
     envs.foldLeft(Map.empty[Symbol.VarSym, Symbol.VarSym])(_ ++ _)
 
   /**
-    * Specializes the given formal parameters `fparams0` w.r.t. the given substitution `subst0`.
-    *
-    * Returns the new formal parameters and an environment mapping the variable symbol for each parameter to a fresh symbol.
+    * Specializes `fparams0` w.r.t. `subst0` and returns a mapping from variable symbols to fresh
+    * variable symbols.
     */
   private def specializeFormalParams(fparams0: List[LoweredAst.FormalParam], subst0: StrictSubstitution)(implicit root: LoweredAst.Root, flix: Flix): (List[MonoAst.FormalParam], Map[Symbol.VarSym, Symbol.VarSym]) = {
-    // Return early if there are no formal parameters.
-    if (fparams0.isEmpty)
```

Comment:
I suppose you have deemed this an unneeded optimization.

Reply:
true, a bit unrelated to the PR, but calling two functions on Nil seems very cheap